Mathlib/Topology/Instances/CantorSet.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2024 Jana Göken. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Artur Szafarczyk, Suraj Krishna M S, JB Stiegler, Isabelle Dubois,
+Tomáš Jakl, Lorenzo Zanichelli, Alina Yan, Emilie Uthaiwat, Jana Göken
+under guidance of Filippo A. E. Nuccio
+-/
+import Mathlib.Topology.Metrizable.Basic
+import Mathlib.Topology.Algebra.GroupWithZero
+import Mathlib.Topology.Instances.Real
+
+/-!
+# Ternary Cantor Set
+This file contains the definition of the Cantor ternary set
+as well as some first properties, which will be updated later.
+
+## Main Definitions
+We give a definition for the ternary Cantor set in this file, which
+is `cantorSet`.
+-/
+
+/-- We define the preCantorSet as the preimages under the iterations of some functions. -/
+def preCantorSet : ℕ → Set ℝ
+  | 0 => Set.Icc 0 1
+  | Nat.succ n => (· / 3) '' preCantorSet n ∪ (fun x ↦ (2 + x) / 3)  '' preCantorSet n
+
+/-- We define the Cantor set as the limit of all preCantorSets. -/
+def cantorSet := iInf preCantorSet
+
+
+
+/-!
+## Simple Properties
+-/
+
+/-- The ternary Cantor set inherits the metric and in particular the topology from the reals. -/
+instance cantorSet.metricSpace : MetricSpace cantorSet := Subtype.metricSpace
+
+
+lemma quarters_mem_preCantorSet (n : ℕ) : 1/4 ∈ preCantorSet n ∧ 3/4 ∈ preCantorSet n := by
```

Comment:
The fact of the matter is that this can't really be split because the proof goes inductively to show that 1/4 is in ``preCantorSet n`` we need that 3/4 is in `preCantorSet (n-1)`` and vice versa.